src/app/core/store/documents/documents.reducer.ts

```diff
@@ -51,6 +61,33 @@ export function documentsReducer(
   }
 }
 
+function patchDocument(state: DocumentsState, action: DocumentsAction.PatchData): DocumentsState {
+  const originalDocument = state.entities[action.payload.document.id];
+  if (!action.payload.originalDocument) {
+    action.payload.originalDocument = originalDocument;
+  }
+
+  return documentsAdapter.upsertOne(
+    {
+      ...action.payload.document,
+      data: {
+        ...originalDocument.data,
+        ...action.payload.document.data,
+      },
+    },
+    state
+  );
+}
+
+function updateDocument(state: DocumentsState, action: DocumentsAction.UpdateData): DocumentsState {
+  const originalDocument = state.entities[action.payload.document.id];
+  if (!action.payload.originalDocument) {
+    action.payload.originalDocument = originalDocument;
```

Comment:
Why do you change the action payload here? Reducer function must not have any side effects!